andela/activo-api
# #159203295 Revisit the search asset endpoints

#### What does this PR do?
Improve the search asset endpoint

#### Description of Task to be completed?
* create an immutableMultiDict to mock request.args
* clean up the dynamic filter by removing the logic for search asset
* modify the search asset endpoint to improve the code for readability

#### How should this be manually tested?
* pull from this branch ```bg-revisit-the-search-asset-endpoints-159203295``` into a new branch
* run ``` pip install -r requrements.txt ```
* endpoint should look like this: ```/assets/search?start=<date>&end=<date>&warranty_start=<date>&warranty_end=<date>```
*date format should be ```Y-M-D```
* run ```pytest ``` to run the test

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[159203295](https://www.pivotaltracker.com/story/show/159203295)

#### Postman collection:
- [Link](https://www.getpostman.com/collections/2f53fc56d99f1ae2f2c8)

#### Screenshots (if appropriate)
<img width="1135" alt="screen shot 2018-07-11 at 7 53 31 am" src="https://user-images.githubusercontent.com/24475008/42555273-975960ba-84df-11e8-94d1-936ae7f37e5f.png">

<img width="727" alt="screen shot 2018-07-23 at 11 43 48 am" src="https://user-images.githubusercontent.com/24475008/43072368-beae6cc2-8e6d-11e8-97f4-9d53b4c1891c.png">

#### Questions:
N/A
